[Feature] Logs Page

BboyAkers opened the pull request:
## Ticket
[STUDIO-160](https://harperdb.atlassian.net/browse/STUDIO-160)

## Overview
- Added a `/Logs` sub-route for instances to view server logs
- Added filtering capabilities for logs

BboyAkers committed:
Merge pull request #454 from HarperDB/stage

v4.8.13

BboyAkers committed:
create logs page

BboyAkers committed:
parsed the logs and put them in a code preview tag

BboyAkers committed:
fixed linting issues

BboyAkers committed:
added row clicking

BboyAkers committed:
created modal on row click

BboyAkers committed:
updated log table view and styling/rendering

BboyAkers committed:
added styling for Log info modal, added filters to readLog api call

BboyAkers committed:
added filter inputs to logs and filter object to readLog call

BboyAkers committed:
basic filtering works

BboyAkers committed:
filtering logs work

BboyAkers committed:
set default form state

BboyAkers committed:
fixed form reset and refactored select elements to use <Select />

BboyAkers committed:
css styling for logs filters header

BboyAkers committed:
fixed themes for log filters form

kriszyp reviewed (commented):
Great job, this is a huge improvement over the previous log viewer.
I will share some of my own (mostly subjective) preferences for log viewing. First, (maybe a little more objective), reading the body of a log message in log info dialog in dark mode is pretty hard:
![image](https://github.com/user-attachments/assets/93203101-2500-4c1e-9f1a-1dc0664785c8)

Generally, the act of log viewing generally involves looking through a large amount of log information and quickly scanning for key bits of information. Consequently, good/mature log viewing tools generally feature high information density, balanced with text legibility. However, in the studio log viewer, much of the screen real estate is "wasted":
![image](https://github.com/user-attachments/assets/a56f9bed-571c-4961-a895-877352cbb686)
The first five columns take up most of the horizontal space even though they are small bits of text. The log message is by far the most important part of the log entry, but it receives a minority of the space. The columns should be adjusted so that the Message column uses about 60%-70% of the horizontal space.
Vertically, the space usage is sub-optimal as well, with the actual content often compromising a small amount of the available space, with font size that is probably a little small, and too much line spacing. I would think you typically want the text height to at least be 50% of the line height. Here is an example of typical log viewing from a terminal that I would consider highly efficient:
![image](https://github.com/user-attachments/assets/52cd42c7-86a8-42e6-bf06-d03fa6a7846d)
Also, why does the scrollbar have to be so tiny for dragging (scrolling is often the first thing I do when I am looking at logs)?
![image](https://github.com/user-attachments/assets/d2a03e65-e033-4b05-8d16-754f68e20288)

Anyway, some of this is probably beyond the scope of what you were aiming to tackle with this PR, and some might be somewhat preferential. You should probably fix the Log Info dialog in dark mode for this PR, but beyond that, I am fine with merging this PR and doing further log updates at a different time, if you prefer.

Again, this is a great step forward, nice work!

BboyAkers committed:
fixed modal message color in dark mode, changed row sizing in large screens

BboyAkers committed:
fixed scrollbar width

BboyAkers commented:
Fixed:
- Log Row column widths/sizing
- Scrollbar width
- Increased height of each Log Row to show more of the message.
- Fixed Log Modal Message text color in dark mode

@kriszyp @mihaiharperdb

kriszyp commented:
It looks like the rows are fixed size now, taking too much space for small space and overflowing for large messages:
![image](https://github.com/user-attachments/assets/23658b54-5bb0-4673-9fea-9634144beeae)

BboyAkers committed:
fixed log limit filter

BboyAkers committed:
fixed message sizing

BboyAkers commented:
Update:
- Log Limit selection issue: Fixed
- Log Level "Info": Should be fixed as well! If not, DM me @mihaiharperdb and we can hop on a call so we can dive into it and figure it out. :slightly_smiling_face:
- Table Row overflow issue: Fixed
- `datetime-local` Input Calendar UI for FireFox: Firefox doesn't support HH:MM:SS in the default calendar UI. It defaults to a text input for that section of the input according to MDN. I can look into importing a Calendar UI dependency in a follow-up PR. I does however work as expected with the full HH:MM:SS UI for:

BboyAkers committed:
updated row sizing

BboyAkers committed:
update logs row height to be the size of the content
